pycqed/scripts/Spectroscopy.py

```diff
@@ -0,0 +1,281 @@
+                             'fit_guess_fn': fit_guess_fn,
+                             'fit_yvals': self.plot_amp,
+                             'fit_xvals': {'f': self.plot_frequency}
+                             }
+        else:
+            self.fit_dict = {'fit_fn': fit_fn,
+                             'fit_guess_fn': fit_guess_fn,
+                             'fit_yvals': [np.squeeze(tt) for tt in self.plot_amp],
+                             'fit_xvals': np.squeeze([{'f': tt[0]} for tt in self.plot_frequency])}
+
+    def do_subtract_background(self,thres=None,back_dict=None,):
+        if len(self.data_dict['timestamps']) == 1:
+            pass
+        else:
+            x_filtered = []
+            y_filtered = []
+            for tt in range(len(self.data_dict['timestamps'])):
+                y = np.squeeze(self.plot_amp[tt])
+                x = np.squeeze(self.plot_frequency)[tt]
+                # print(self.plot_frequency)
+                # [print(x.shape) for x in self.plot_frequency]
+                # print(x)
+                # print(y)
+                # print(len(x),len(y))
+                guess_dict = SlopedHangerFuncAmplitudeGuess(y, x)
+                Q = guess_dict['Q']['value']
+                f0 = guess_dict['f0']['value']
+                df = 2*f0/Q
+                fmin = f0-df
+                fmax = f0+df
+                indices = np.logical_or(x < fmin*1e9, x > fmax*1e9)
+                x_filtered.append(x[indices])
+                y_filtered.append(y[indices])
+            self.background = pd.concat([pd.Series(y_filtered[tt], index=x_filtered[tt])
+                       for tt in range(len(self.data_dict['timestamps']))], axis=1).mean(axis=1)
+            background_vals = self.background.reset_index().values
+            freq = background_vals[:,0]
+            amp = background_vals[:,1]
+            # thres = 0.0065
+            indices = amp < thres
+            freq = freq[indices]*1e-9
+            amp = amp[indices]
+            fit_fn = double_cos_linear_offset
+            model = lmfit.Model(fit_fn)
+            fit_yvals = amp
+            fit_xvals = {'t':freq}
+            # fit_guess_fn = double_cos_linear_offset_guess
+            # guess_dict = fit_guess_fn(fit_yvals, **fit_xvals)
+            for key, val in list(back_dict.items()):
+                    model.set_param_hint(key, **val)
+            params = model.make_params()
+            print(fit_xvals)
+            fit_res = model.fit(fit_yvals,
+                                 params=params,
+                                 **fit_xvals)
+            self.background_fit = fit_res
+
+            for tt in range(len(self.data_dict['timestamps'])):
+                divide_vals = fit_fn(np.squeeze(self.plot_frequency)[tt]*1e-9,**fit_res.best_values)
+                self.plot_amp[tt] = np.array([np.array([np.divide(np.squeeze(self.plot_amp[tt]),divide_vals)])]).transpose()
+
+
+
+    def plot_fitting(self):
```

Comment:
I specified these plots in the same way as normal plots using the plotting dicts. 
I will try to make my example a bit more clear but I think that method is better as it enforces the same paradigm everywhere.